### ISSUE TITLE
rust: declare generated functions as inline able

### ISSUE THREAD

**Issue by crop2000** [grame-cncm/faust#1093]
I think it should make sense to add a `#[inline]` declaration to most functions generated by faust.
here a example how the generated code would look like:
[example](https://gist.github.com/crop2000/22960fa5713b78037ba61ee822aa49a8)

The main benefit of explicit inline declaration is that functions can be in-lined across crate boundaries.

there are some things to consider in relation to this as discussed here:
https://matklad.github.io/2021/07/09/inline-in-rust.html
the small examples and faust2* projects should not be affected by this change as far as I understood.

do you have opinions on this?
@bluenote10

**Comment by crop2000** [comment]
this pr shows a solution (and a discussion of the problem)
https://github.com/rust-lang/hashbrown/pull/119/files
Similar to our use of default-boxed it this should lead to compiler warnings if the Cargo.toml isn't prepared for that.

**Comment by sletz** [comment]
In practice, what kind of functions are called "often enough" to justify inlining ? `get_param/set_param/compute` possibly ? Most of the others are called like once at init time or a few times like `build_user_interface`.  

Any concrete use case of the "inlining across crate boundaries"  you can see ?

**Comment by crop2000** [comment]
Yes,especially  get and set parameters and compute. The inline flag does not force inlining but makes it possible.
